### ISSUE TITLE
Warning: Undefined array key "var" with Domain Theme submodule

### ISSUE THREAD

**Issue by stpaultim** [backdrop-contrib/domain#43]
`Warning: Undefined array key "var" in domain_theme_settings_submit() (line 305 of /var/www/html/modules/domain/domain_theme/domain_theme.admin.inc).`

Getting this error when trying to assign a theme to a specific domain with the domain_theme sub-module enabled. The modules works, but generates this warning.

**Comment by stpaultim** [comment]
This is another warning that may be related.

`Warning: Undefined array key "favicon_upload" in domain_theme_file_upload() (line 404 of /var/www/html/modules/domain/domain_theme/domain_theme.admin.inc).`

**Comment by stpaultim** [comment]
@sudipto68 and I were looking at these errors tonight and determined that they did not come up during some basic use of the module. 

To recreate these errors, I had to assign both domains the Basis theme and then use the Color Module to change the header color. This triggered these specific errors. 

I'm not sure I could consider these "Warnings" to block a beta release.

Once again, even in these conditions, the module works as expected, other than generating these warnings.

**Comment by izmeez** [comment]
My testing of the domain_theme submodule yields failures and successes.

Steps: On the default domain enable themes that may be used at `/admin/appearance/list`. Go to the domain list at `/admin/structure/domain/view`.  Choose a subdomain, click on `Edit domain` and click on tab menu item `Theme`. A different theme can be selected and then click `Set domain theme`. This works and the new theme shows for the chosen subdomain without displaying errors or warnings on php 7.4.x

However, efforts to configure a theme, such as changing the header color in basis, fail using the above process or alternatively:

Go to a subdomain url. Go to appearance and click on settings for theme. A notice warns to use the url for the specific subdomain `/admin/structure/domain/view/3/theme/basis/theme-settings`. Click on suggested url. Then modify the theme, eg, header color. Note: the submit button says: `Save Theme Settings`. When user clicks on this, in our case, 3 warnings are displayed:

>Notice: Undefined index: logo_upload in domain_theme_file_upload() (line 397 of \modules\domain\domain_theme\domain_theme.admin.inc).
>Notice: Undefined index: favicon_upload in domain_theme_file_upload() (line 404 of \modules\domain\domain_theme\domain_theme.admin.inc).
>Notice: Undefined index: var in domain_theme_settings_submit() (line 305 of \modules\domain\domain_theme\domain_theme.admin.inc).

AND the screen shows a list of themes and the submit button at the bottom says: `Set domain theme` as with previous method above. Whether or not user clicks on that submit button, returning to the site shows the change has not taken effect.

So I am not able to replicate your findings as described.

I have only enabled the `domain` and `domain_theme` modules. Maybe, you also have the `domain_settings` module enabled?

**Comment by izmeez** [comment]
Possibly, the reason for the differences in my findings are because I had already applied the fix in #39 and the PR #46.

**Comment by sudipto68** [comment]
I have added a PR [#67](https://github.com/backdrop-contrib/domain/pull/67) for this warning and some related warnings. Can anyone please check the [code](https://github.com/backdrop-contrib/domain/pull/67/commits/c0f7f43a54233dc57cfbe6b76b66797ce15c8792) and also test if this is working?

**Comment by yorkshire-pudding** [comment]
@sudipto68 @stpaultim - I have reproduced these warnings and can confirm that the patch gets rid of the warnings and the code looks mostly ok from my perspective, though a couple of coding standard changes needed - see PR.

I noticed while testing this, that it does not appear to be possible to have different Basis configurations on different domains. How common that need is I really don't know; I would be comfortable in that being known limitation if need be.  In most of my testing so far I've used completely different themes for each domain so is not an issue for me

**Comment by stpaultim** [comment]
@yorkshire-pudding I agree, that we can live with the limitation that Domain can not use different configuration for Basis on different domains.

**Comment by stpaultim** [comment]
@sudipto68 - Here is some coding feedback I got today. I don't think this is urgent, but another way of accomplishing the same thing. 

Instead of:
`  if (array_key_exists('var', $values)) {
    $key = $values['var'];
  }`

Maybe this:
`!empty($values['var']) {}`

Let's revisit this later.
